sdk/core/azure_core/src/lib.rs

```diff
@@ -31,3 +30,10 @@ pub use typespec_client_core::{
 
 #[cfg(feature = "xml")]
 pub use typespec_client_core::xml;
+
+pub use typespec_client_core::get_async_runtime;
+pub use typespec_client_core::set_async_runtime;
+
+pub mod async_runtime {
+    pub use typespec_client_core::async_runtime::{AsyncRuntime, SpawnedTask, TaskFuture};
+}
```

Comment:
You could just replace this with:
```suggestion
 pub use typespec_client_core::async_runtime;
```

Reply:
Makes sense. I was primarily worried about types inside async_runtime leaking out into azure_core and typespec_client_core. 

I also moved all of this into `azure_core::async_runtime` rather than over promoting.

Reply:
Those are equivalent. `typespec_client_core` should be a subset of `azure_core` anyway, and I want to try to keep the structure the same. Just makes everything easier. I did this same change with some other modules e.g., `http::response` where I didn't need to merge anything local.